smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -64,12 +66,12 @@ contract MixinPurchase is
     if (toKey.expirationTimestamp >= block.timestamp) {
       // This is an existing owner trying to extend their key
       toKey.expirationTimestamp = toKey.expirationTimestamp.add(expirationDuration);
-      emit ExpirationChanged(toKey.tokenId, expirationDuration, true);
+      emit RenewKeyPurchase(_recipient, toKey.expirationTimestamp.add(expirationDuration));
     } else {
       // SafeAdd is not required here since expirationDuration is capped to a tiny value
       // (relative to the size of a uint)
       toKey.expirationTimestamp = block.timestamp + expirationDuration;
-      emit ExpirationChanged(toKey.tokenId, expirationDuration, true);
+      emit RenewKeyPurchase(_recipient, block.timestamp + expirationDuration);
```

Comment:
Hmm I'm just noticing this will emit on fist purchase - right?

Reply:
Dang it. yes it will. but of course we don't want it to.

Reply:
The flow for a new (first) purchase needs to skip this entire `if/else` block

Reply:
yea - we have 3 use cases here.  i think we can if elseif else it.  1) new purchase 2) extend a non-expired key 3) repurchase an expired key

Reply:
yup. copy that.